File: Data Structures & Algorithms/concatenation-of-array/submission-0.java
class Solution {
    public int[] getConcatenation(int[] nums) {
        if (nums.length == 0) {
            return nums;
        }

        int[] ans = new int[nums.length * 2];
        for (int i = 0; i < nums.length; i++) {
            ans[i] = nums[i];
            ans[i+nums.length] = nums[i];
        }
        return ans;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        # 2 3 4 5 1
        l, r = 0, len(nums) - 1
        if nums[l] <= nums[r]:
            return nums[l]
        
        while (r - l) > 1:
            mid = (r + l) // 2
            num = nums[mid];

            if num > nums[l]:
                l = mid
            else:
                r = mid
        
        return nums[r]



        

